core/templates/pages/translate-custom-parser.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Custom parser for translations.
+ */
+
+import { TranslateDefaultParser, TranslateParser } from '@ngx-translate/core';
+import constants from 'assets/constants';
+import MessageFormat from 'messageformat';
+import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+
+export class TranslateCustomParser extends TranslateParser {
+  messageFormat;
+  constructor(
+    private translateDefaultParser: TranslateDefaultParser,
+    private i18nLanguageCodeService: I18nLanguageCodeService
+  ) {
+    super();
+    this.translateDefaultParser.templateMatcher = /<\[\s?([^{}\s]*)\s?\]>/g;
```

Comment:
The default template looks for `{{ }}` while interpolating.
But we use `<[ ]>` in our translations. This was done to avoid conflict with jinja

This regex looks for `<[ ]>` for interpolation.

![image](https://user-images.githubusercontent.com/24855641/118940708-9ad31d00-b96e-11eb-8022-68bbfbd4c339.png)

Reply:
Thanks for the explanation! :)